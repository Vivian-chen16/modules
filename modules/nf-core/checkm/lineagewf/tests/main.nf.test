nextflow_process {

    name "Test Process CHECKM_LINEAGEWF"
    script "../main.nf"
    process "CHECKM_LINEAGEWF"
    tag "modules"
    tag "modules_nfcore"
    tag "checkm"
    tag "checkm/lineagewf"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "$outputDir"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = [
                                [ id:'test', single_end:false ], // meta map
                                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                            ]
                input[1] = 'fasta'
                input[2] = [] // Downlod CheckM database
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
