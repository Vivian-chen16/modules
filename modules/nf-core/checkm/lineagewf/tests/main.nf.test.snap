{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    "genes.faa:md5,3f1e7174f15de3ba371f06898bb21e1d",
                                    "genes.gff:md5,9bf49592aa6dc175d6f111f56ccfc167",
                                    "hmmer.analyze.txt:md5,cca1179a7e368c31d35dc4b9b40174f0",
                                    "hmmer.tree.txt:md5,cdbb25587f858f21d31410fca824c83c"
                                ]
                            ],
                            "checkm.log:md5,a2a1d900384db6964963441f315abe65",
                            "lineage.ms:md5,4ab1122d999ecf4f25992be27376c8ae",
                            [
                                [
                                    
                                ],
                                "bin_stats.analyze.tsv:md5,4b64c336c726895f1f903f5d0e046252",
                                "bin_stats.tree.tsv:md5,4b64c336c726895f1f903f5d0e046252",
                                "bin_stats_ext.tsv:md5,7326697ed54e8288b94a0502c7c82306",
                                "checkm_hmm_info.pkl.gz:md5,d727d082dd93dfff6f56f4f48544b929",
                                "marker_gene_stats.tsv:md5,e2203d0fd929bdcf2bcdfe7236717036",
                                "phylo_hmm_info.pkl.gz:md5,d5d95ba3c47913634cb823ca7b956c13",
                                [
                                    "concatenated.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                                    "concatenated.tre:md5,08efbb5a0b56a76d5241936fa588ee44"
                                ]
                            ]
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "lineage.ms:md5,4ab1122d999ecf4f25992be27376c8ae"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d5559764f563c4b55223e4e4a3dc1ec9"
                    ]
                ],
                "3": [
                    "versions.yml:md5,fd7a8db316717ab0695e47d7d0a645ff"
                ],
                "checkm_output": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    "genes.faa:md5,3f1e7174f15de3ba371f06898bb21e1d",
                                    "genes.gff:md5,9bf49592aa6dc175d6f111f56ccfc167",
                                    "hmmer.analyze.txt:md5,cca1179a7e368c31d35dc4b9b40174f0",
                                    "hmmer.tree.txt:md5,cdbb25587f858f21d31410fca824c83c"
                                ]
                            ],
                            "checkm.log:md5,a2a1d900384db6964963441f315abe65",
                            "lineage.ms:md5,4ab1122d999ecf4f25992be27376c8ae",
                            [
                                [
                                    
                                ],
                                "bin_stats.analyze.tsv:md5,4b64c336c726895f1f903f5d0e046252",
                                "bin_stats.tree.tsv:md5,4b64c336c726895f1f903f5d0e046252",
                                "bin_stats_ext.tsv:md5,7326697ed54e8288b94a0502c7c82306",
                                "checkm_hmm_info.pkl.gz:md5,d727d082dd93dfff6f56f4f48544b929",
                                "marker_gene_stats.tsv:md5,e2203d0fd929bdcf2bcdfe7236717036",
                                "phylo_hmm_info.pkl.gz:md5,d5d95ba3c47913634cb823ca7b956c13",
                                [
                                    "concatenated.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                                    "concatenated.tre:md5,08efbb5a0b56a76d5241936fa588ee44"
                                ]
                            ]
                        ]
                    ]
                ],
                "checkm_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d5559764f563c4b55223e4e4a3dc1ec9"
                    ]
                ],
                "marker_file": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "lineage.ms:md5,4ab1122d999ecf4f25992be27376c8ae"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fd7a8db316717ab0695e47d7d0a645ff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-21T19:31:18.52852479"
    }
}